.github/actions/build-and-cache-rust-tool/action.yml

```
@@ -0,0 +1,57 @@
+name: 'Build and cache random Rust tools'
```

I guess you want maybe to use `on.workflow_call` for creating a reusable workflow.
https://docs.github.com/en/actions/using-workflows/reusing-workflows#creating-a-reusable-workflow

```suggestion
on:
  workflow_call:
name: 'Build and cache random Rust tools'
```

---

Maybe. But after reading the error message more closely it tells me that the `shell` variable is missing.

---

Seems to work now. Just want to check if we get a problem on windows in the future

---

It seems there will be a problem on windows but not due to this refactoring but the path for the upload of the artifact